tests/drivers/fprotect/positive/src/main.c

```diff
@@ -23,7 +23,7 @@ static uint32_t control_read_addr = 0x6032;
 static uint8_t read_data_before[sizeof(write_data)];
 static uint8_t read_data_after[sizeof(write_data)];
 
-static void test_flash_write(void)
+ZTEST(test_fprotect_positive, test_flash_awrite)
```

Comment:
I'm not a big fan of this solution. It's quite tricky to name it as "awrite" only to enforce test order. If you really need to do this in this way, please name tests in more meaningful way, like `test_1_...`, `test_2_...`, etc.  In this case please move this `flash_write` to test suite setup, as for example:
```c
static void *setup(void)
{
	int retval = flash_read(flash_dev, control_read_addr, read_data_before,
				ARRAY_SIZE(read_data_before));

	zassert_true(retval == 0, "flash_read failed");
	retval = flash_write(flash_dev, valid_write_addr, write_data,
			     ARRAY_SIZE(write_data));
	zassert_true(retval == 0, "flash_write failed");

	return NULL;
}

ZTEST_SUITE(test_fprotect_positive, NULL, setup, NULL, NULL, NULL);
```

Reply:
I think it should stay as a test. It is a test, it just has side effects that affect the next two tests. It's a test that deals with flash, after all.